service/history/api/get_workflow_util.go

```diff
@@ -94,6 +96,10 @@ func GetOrPollMutableState(
 	// We return the full version histories. Callers need to fetch the last version history item from current branch
 	// and use the last version history item in following calls.
 	if !versionhistory.ContainsVersionHistoryItem(currentVersionHistory, request.VersionHistoryItem) {
+		logger.Warn("Request history branch and current history branch are mismatched.",
+			tag.Value(currentVersionHistory),
```

Comment:
I don't know much about this yet; is `currentVersionHistory` safe to log? Safe in terms of (a) security, (b) size and (c) performance?

Reply:
change to log the last item. The length is fixed.